make sysprep fails on macOS 10.12.x

Running `make sysprep` fails with the following error on a clean system:


```
make sysprep
[+] found darwin provision script: /Users/nwalck/code/osquery/tools/provision/darwin.sh
[+] checking for updates to brew
HEAD is now at b55290849 Merge pull request #2851 from MikeMcQuaid/missing-formula-no-core
[+] refreshing local tap: osquery-local
[+] installing and updating Homebrew core
==> Tapping homebrew/core
Cloning into '/usr/local/osquery/Library/Taps/homebrew/homebrew-core'...
remote: Counting objects: 369849, done.
remote: Compressing objects: 100% (43/43), done.
remote: Total 369849 (delta 46), reused 54 (delta 31), pack-reused 369775
Receiving objects: 100% (369849/369849), 127.39 MiB | 2.81 MiB/s, done.
Resolving deltas: 100% (217675/217675), done.
Error: Invalid formula: /usr/local/osquery/Library/Taps/homebrew/homebrew-core/Formula/go.rb
undefined method `head?' for nil:NilClass
Error: Cannot tap homebrew/core: invalid syntax in tap!
make: *** [sysprep] Error 1
```

This is resolved by manually cloning homebrew-core into /usr/local/osquery/Library/Taps/homebrew/:

```
mkdir -p /usr/local/osquery/Library/Taps/homebrew
git clone https://github.com/Homebrew/homebrew-core.git /usr/local/osquery/Library/Taps/homebrew/homebrew-core
```

After doing the above clone, `make sysprep` runs as expected.

--- a/tools/provision.sh
+++ b/tools/provision.sh
@@ -19,9 +19,9 @@ LINUXBREW_REPO="https://github.com/Linuxbrew/brew"
 
 # Set the SHA1 commit hashes for the pinned homebrew Taps.
 # Pinning allows determinism for bottle availability, expect to update often.
-HOMEBREW_CORE="f7e573dfdbe26f00bf39d6dd8244d253ca4c1a94"
+HOMEBREW_CORE="941ca36839ea354031846d73ad538e1e44e673f4"
 LINUXBREW_CORE="ff436960b24eb7f283d91aee578606b4995a52d2"
-HOMEBREW_BREW="b5529084906af89827f6d9befd613457a1615918"
+HOMEBREW_BREW="ac2cbd2137006ebfe84d8584ccdcb5d78c1130d9"
 LINUXBREW_BREW="269be25411af00b5ecf9764a92c8061dc79ed8ca"
 
 # If the world needs to be rebuilt, increase the version

